### ISSUE TITLE
parameter for maximum distance

### ISSUE THREAD

**Issue by bittlingmayer** [sindresorhus/leven#14]
When testing many for or sorting by similarity, often we are happy to short-circuit on any pair with a distance greater than `max`.

`leven(a, b, 3)`

At scale, given that the least similar strings are also very expensive to compute, there is a big potential savings.  (For one, just comparing length is enough to discard many candidates.)

(For such pairs, the distance can be `null`, `max` or `max + 1` - I don't have a strong opinion on which is best.)

**Comment by sindresorhus** [comment]
Makes sense. PR welcome. Should be an options-object with an option called `maximumDistance`.

**Comment by sindresorhus** [comment]
> but probably max is the most useful, as it would require not extra handling for sorting or filtering.)

👍

**Comment by Yomguithereal** [comment]
Feel free to take inspiration from this code: https://github.com/Yomguithereal/talisman/blob/master/src/metrics/distance/levenshtein.js#L230-L340

**Comment by ka-weihe** [comment]
This is not the fastest Levenshtein implementation for NodeJS. In fact it is not even the second fastest. The fastest is:
https://github.com/ka-weihe/node-levenshtein

**Comment by sindresorhus** [comment]
If anyone wants to work on this, see https://github.com/sindresorhus/leven/pull/15 for the previous attempt and the feedback there.
